dask/dataframe/io/parquet/arrow.py

```diff
@@ -119,6 +120,51 @@ def _determine_dataset_parts(fs, paths, gather_statistics, filters, dataset_kwar
     return parts, dataset
 
 
+def _write_partitioned(table, root_path, partition_cols, filesystem=None, **kwargs):
+    """ Write table to a partitioned dataset with pyarrow.
+
+        Logic copied from pyarrow.parquet.
+        (arrow/python/pyarrow/parquet.py::write_to_dataset)
+    """
+    fs, root_path = pq._get_filesystem_and_path(filesystem, root_path)
+    pq._mkdir_if_not_exists(fs, root_path)
```

Comment:
the downside of copying the code here is that in principle, those methods are private, and thus not guaranteed to be stable ..

Reply:
Why not use `fsspec` methods here? `filesystem` should already be a `fsspec` one here, so `filesystem.mkdirs(root_path, exists_ok=True` will work.

Reply:
Yes, I think those two lines should be rather easy to replace with general  / public API.

Reply:
@martindurant - I'm getting `TypeError: mkdirs() got an unexpected keyword argument 'exists_ok'`

Reply:
sorry, should be `exist_ok`. Also, `makedirs()` is perhaps more canonical (the former aliases the latter).